backend/fms_core/template_importer/importers/sample_add_metadata.py

```diff
@@ -0,0 +1,55 @@
+from ._generic import GenericImporter
+from fms_core.template_importer.row_handlers.sample_add_metadata import SampleAddMetadataHandler
+from fms_core.templates import SAMPLE_ADD_METADATA_TEMPLATE
+
+METADATA_STARTING_INDEX = 4
+
+class SampleAddMetadataImporter(GenericImporter):
+    SHEETS_INFO = SAMPLE_ADD_METADATA_TEMPLATE['sheets info']
+
+    def __init__(self):
+        super().__init__()
+        self.metadata_starting_index = METADATA_STARTING_INDEX
+
+    def import_template_inner(self):
+        """
+            METADATA SHEET
+        """
+        metadata_sheet = self.sheets['Metadata']
+
+        # Add the extra headers to the template sheet according to the metadata
+        metadata_headers = [key for i, (key, val) in enumerate(metadata_sheet.rows[0].items()) if i >= self.metadata_starting_index]
+        extra_headers = {
+            # sheet_number : extra metadata headers
+            0: metadata_headers
+        }
+        self.add_dynamic_headers(extra_headers)
+
+        # Iterate through experiment rows
+        for row_id, row in enumerate(metadata_sheet.rows):
+            row_data = {}
+            metadata = {}
+            for i, (key, val) in enumerate(row.items()):
+                if i < self.metadata_starting_index:
+                    row_data[key] = row[key]
+                # Just add metadata if the cell is not empty, else ignore
+                elif row[key]:
```

Comment:
i would enforce lower case on the name of the metadata ... i guess this is where this would happen. This is to prevent having to test in the model if the name match another with different letter case. We could also create a new "lowercase name" validator to make sure we do not insert upper case names by a back door.